src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -663,8 +663,79 @@ case object LOG3 extends LogOp(0xa3)
 case object LOG4 extends LogOp(0xa4)
 
 
-sealed abstract class CallOp(code: Int, delta: Int, alpha: Int) extends OpCode(code, delta, alpha, _.G_zero) {
+case object CREATE extends OpCode(0xf0, 3, 1, _.G_create) {
+  protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
+    val (Seq(endowment, inOffset, inSize), stack1) = state.stack.pop(3)
+
+    val validCall = state.env.callDepth < EvmConfig.MaxCallDepth && endowment <= state.ownBalance
 
+    if (!validCall) {
+      val stack2 = stack1.push(UInt256.Zero)
+      state.withStack(stack2).step()
+    } else {
+
+      val (initCode, memory1) = state.memory.load(inOffset, inSize)
+      val (newAddress, world1) = state.world.newAddress(state.env.ownerAddr)
+      val world2 = world1.transfer(state.env.ownerAddr, newAddress, endowment)
+
+      val newEnv = state.env.copy(
+        callerAddr = state.env.ownerAddr,
+        ownerAddr = newAddress,
+        value = endowment,
+        program = Program(initCode),
+        inputData = ByteString.empty,
+        callDepth = state.env.callDepth + 1
+      )
+
+      //to avoid calculating this twice, we could adjust state.gas prior to execution in OpCode#execute
+      //not sure how this would affect other opcodes
+      val availableGas = state.gas - (constGasFn(state.config.feeSchedule) + varGas(state))
+      val startGas = state.config.gasCap(availableGas)
+
+      val context = ProgramContext[W, S](newEnv, startGas, world2, state.config)
+      val result = VM.run(context)
+
+      val codeDepositGas = state.config.calcCodeDepositCost(result.returnData)
+      val gasUsedInVm = startGas - result.gasRemaining
+      val totalGasRequired = gasUsedInVm + codeDepositGas
+      val enoughGasForDeposit = totalGasRequired <= availableGas
+
+      if (result.error.isDefined) {
+        val stack2 = stack1.push(UInt256.Zero)
+        state.withStack(stack2).spendGas(startGas)
```

Comment:
`availableGas` is the current machine gas minus the part that is spent [here](https://github.com/input-output-hk/etc-client/blob/master/src/main/scala/io/iohk/ethereum/vm/OpCode.scala#L178). This the gas available to execute this instruction and all the remaining program.

`startGas` is a fraction of `availableGas` and it is the maximum value that can be used by the initialisation code, which runs in the recursively spawned child VM. If that code runs correctly most likely less gas will be used, but if it results in an error we need to consume all of `startGas`.

To be exact `startGas == availableGas * 63/64`. In the YP it is defined in the `CREATE` opcode definition. See the 4th argument to the _Λ_ function, i.e. _L(μ<sub>g</sub>)_, and the _L_ function is defined in the eq (224).

Note that before EIP-150 it was `startGas == availableGas`.